packages/backend-common/src/database/config.ts

```diff
@@ -23,5 +23,5 @@ import { merge } from 'lodash';
  * @param overrides Any additional overrides
  */
 export function mergeDatabaseConfig(config: any, ...overrides: any[]) {
-  return merge(config, ...overrides);
+  return merge({}, config, ...overrides);
```

Comment:
@freben @andrewthauer I think this would properly fix it for all the database code, is this better?

Reply:
Likely yep. You should be able to write a test to confirm.

Reply:
Test implemented, please have a look.